scripts/test-3dsmax-adapter-run.ps1

```
@@ -0,0 +1,417 @@
+        return $null
+    }
+    
+    # Parse YAML properly
+    $paramContent = Get-Content $paramFile -Raw
+    $yamlData = ConvertFrom-Yaml $paramContent
+    
+    # Extract values from parsed YAML
+    $getValue = { param($name) 
+        $param = $yamlData.parameterValues | Where-Object { $_.name -eq $name }
+        if ($param) { 
+            return $param.value 
+        } else { 
+            return "" 
+        }
+    }
+    
+    # Get frames for run data
+    $frames = & $getValue "Frames"
+    $frames = if ([string]::IsNullOrEmpty($frames)) { "0" } else { $frames }
+    
+    # Read template file to extract renderer, state_set, and output_file_name
+    $templateFile = Join-Path $JobBundleDir "template.yaml"
+    if (-not (Test-Path $templateFile)) {
+        Write-Error "Template file not found: $templateFile"
+        return $null
+    }
+    
+    $templateContent = Get-Content $templateFile -Raw
+    $templateData = ConvertFrom-Yaml $templateContent
+    
+    # Validate step number against available steps
+    if (-not $templateData.steps -or $templateData.steps.Count -eq 0) {
+        Write-Error "No steps found in template file"
+        return $null
+    }
+    
+    if ($Step -lt 0 -or $Step -ge $templateData.steps.Count) {
+        Write-Error "Step number $Step is out of range. Available steps: 0 to $($templateData.steps.Count - 1)"
+        Write-Host "Available steps:" -ForegroundColor Yellow
+        for ($i = 0; $i -lt $templateData.steps.Count; $i++) {
+            Write-Host "  Step $i`: $($templateData.steps[$i].name)" -ForegroundColor Gray
+        }
+        return $null
+    }
+    
+    # Get the selected step
+    $selectedStep = $templateData.steps[$Step]
+    Write-Host "Selected step: $($selectedStep.name)" -ForegroundColor Green
+    
+    # Extract camera from taskParameterDefinitions
+    $defaultCamera = "Unknown"  # fallback
```

Is this a good fallback? Should we just fail if one is not provided?

---

Yes this is a good fallback to force it to fail.